component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java

```diff
@@ -767,6 +709,105 @@ private String getMessageBody(Object payload) {
         }
     }
 
+    void initClientConnector(DynamicOptions dynamicOptions) {
+        if (publisherURLOption.isStatic()) {
+            publisherURL = publisherURLOption.getValue();
+        } else {
+            publisherURL = publisherURLOption.getValue(dynamicOptions);
+        }
+
+        String scheme = HttpSinkUtil.getScheme(publisherURL);
+        this.httpURLProperties = HttpSinkUtil.getURLProperties(publisherURL);
+        //Generate basic sender configurations
+        SenderConfiguration senderConfig = HttpSinkUtil
+                .getSenderConfigurations(httpURLProperties, clientStoreFile, clientStorePass, configReader);
+        if (EMPTY_STRING.equals(publisherURL)) {
+            throw new SiddhiAppCreationException("Receiver URL found empty but it is Mandatory field in " +
+                    "" + HttpConstants.HTTP_SINK_ID + "in" + streamID);
+        }
+        if (HttpConstants.SCHEME_HTTPS.equals(scheme) && ((clientStoreFile == null) || (clientStorePass == null))) {
+            throw new ExceptionInInitializerError("Client trustStore file path or password are empty while " +
+                    "default scheme is 'https'. Please provide client " +
+                    "trustStore file path and password in" + streamID);
+        }
+        //if username and password both not equal to null consider as basic auth enabled if only one is null take it
+        // as exception
+        if ((EMPTY_STRING.equals(userName) ^
+                EMPTY_STRING.equals(userPassword))) {
+            throw new SiddhiAppCreationException("Please provide user name and password in " +
```

Comment:
Log siddhi app name also when you throw SiddhiAppCreationException.

Reply:
Fixed with 8f6f25e